File: config/initializers/aws.rb
Aws.config.update({
  region: ENV["AWS_REGION"],
  credentials: Aws::Credentials.new(ENV["AWS_ACCESS_KEY_ID"], ENV["AWS_SECRET_ACCESS_KEY"]),
  endpoint: "https://s3.amazonaws.com"
})

S3_DATA_BUCKET_NAME = Aws::S3::Resource.new.bucket(ENV["S3_DATA_BUCKET_NAME"])

File: app/javascript/app/services/aws.js
import S3 from 'aws-sdk/clients/s3';

const creds = {
  accessKeyId: process.env.AWS_ACCESS_KEY_ID,
  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
  region: process.env.AWS_REGION
};

export const initS3 = () => {
  S3.config.update(creds);
  return new S3();
};

export const getImageUrl = params => {
  const s3 = initS3();
  return s3.getSignedUrl('getObject', params);
};

export const getBucketObjects = (bucket, callback, prefix) => {
  const s3 = initS3();
  s3.listObjectsV2({ Bucket: bucket, Prefix: prefix }, callback);
};
